### PR TITLE
[PT][FSDP] Combine _utils.py into _common_utils.py [2/2]

### PR DESCRIPTION
**Pull request by AlbertDachiChen**
Summary:
https://github.com/pytorch/pytorch/issues/97813
This diffs moves `_no_dispatch_record_stream` and `_same_storage_as_data_ptr`

Test Plan: CI

Differential Revision: D47706114

### CODE REVIEW

File: torch/distributed/fsdp/_utils.py [hunk 1]
```diff
@@ -1,21 +0,0 @@
-import torch
-
-from torch.utils._mode_utils import no_dispatch
-
-
-def _same_storage(x: torch.Tensor, y: torch.Tensor) -> bool:
```
Comment (awgu) [inline torch/distributed/fsdp/_utils.py:6]: What happened to this function?

Comment (awgu) [reply torch/distributed/fsdp/_utils.py:6]: I recall it might not be used anywhere, but maybe we just keep it for this PR?

Comment (AlbertDachiChen) [reply torch/distributed/fsdp/_utils.py:6]: Yeah, it is not used anywhere. Why are we keeping dead code?

Comment (awgu) [reply torch/distributed/fsdp/_utils.py:6]: It is not a big deal for this one probably, but users tend to still import private functions. It is a good practice to check that no one internally is importing it. I was just thinking to separate it into a different PR to isolate the risk of reverting.

Feel free to keep your PR as is.

**Review by awgu — APPROVED** [review]

**Comment by awgu** [comment]
Failure looks unrelated:

[trunk / win-vs2019-cpu-py3 / test (default, 3, 3, windows.4xlarge.nonephemeral)](https://github.com/pytorch/pytorch/actions/runs/5692454801/job/15430498936)
```
Error: The process cannot access the file 'C:\actions-runner\_work\_actions\pytorch\pytorch\main\aten\src\ATen\native\quantized\cpu\qnnpack\src\u8lut32norm' because it is being used by another process.
```

**Pull request closed without merging**